open-grant-proposals/open-proposal-mrd.md

```diff
@@ -0,0 +1,108 @@
+# RFP Proposal: `DDPFT consensus`
+
+**Name of Project:**
+
+**Link to RFP:** Please link to the RFP that you are submitting a proposal for.
+
+**RFP Category:**  `technical-design`
+
+**Proposer:** [@plotozhu](https://github.com/plotozhu)
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:**
+"YES"
+
+# Project Description
+For retrieve markets of Filecoin, data traffic occurs off chain and results should be submitted to chain. To achieve this, client node should send receipts to retrieving miner on accepted data.
+
+Cause it was designed as every 256KB per chunk and length of receipt should be about 160Bytes, occupying about 0.1% bandwith of total nodes. It would mean that if filecoin had 100T bandwith,  on chain bandwith for receipts  would be about 100G, and tps would reach 100G/160 = 625M, which was unaccpetable.
+
+We present a algorithm call MRD(Mergable Receipts Delivery) to solve this problem.
+
+
+## Value
+With MRD, client node sends receipt to retrieval node on each data chunk, retrieval node collects receipts. Multiple receipts can be merged to one, retrieval miner can merge, pack these receipts and broadcast to chain periodly.
+We have tested this algorithm in thounds of retrieval nodes and millions of clients, and proved it worked well.
```

Comment:
Have you tested this algorithm with thousands of **Filecoin** retrieval nodes? Please say more